src/main/java/org/spongepowered/api/command/CommandLowLevel.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.command;
+
+import org.spongepowered.api.command.specification.CommandSpec;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+/**
+ * A low-level interface for commands that can be executed. For almost all use
+ * cases, higher-level tools should be used instead, like
+ * {@link CommandSpec}.
+ *
+ * <p>Implementations are not required to implement a sane
+ * {@link Object#equals(Object)} but really should.</p>
+ */
+public interface CommandLowLevel {
```

Comment:
> This idea of retaining backwards compatibility is doing this pr far more harm than good IMO.

I'm for keeping backwards compatibility so long as in the implementation there is an example provided of a plugin using both systems, or a transition example from the old system to the new. There's enough breaking changes slated for API that I don't feel it necessary to completely break the command system wholeheartedly when we can avoid more breaking changes between versions (look at deprecations versus flat out breakages).